components/property_info.rst

```diff
@@ -0,0 +1,432 @@
+``addProperty()``, ``removeProperty()``, etc.
+
+.. versionadded:: 2.8
+    The PropertyInfo component was introduced in Symfony 2.8.
+
+.. _`components-property-information-installation`:
+
+Installation
+------------
+
+You can install the component in two different ways:
+
+* :doc:`Install it via Composer</components/using_components>` (``symfony/property-info`` on `Packagist`_);
+* Use the official Git repository (https://github.com/symfony/property-info).
+
+.. include:: /components/require_autoload.rst.inc
+
+Additional dependencies may be required for some of the
+:ref:`extractors provided with this component<components-property-info-extractors>`.
+
+.. _`components-property-information-usage`:
+
+Usage
+-----
+
+The entry point of this component is a new instance of the
+:class:`PropertyInfoExtractor<Symfony\\Component\\PropertyInfo\\PropertyInfoExtractor>` class, providing sets of
+information extractors.
+
+.. code-block:: php
+
+    use Symfony\Component\PropertyInfo\PropertyInfoExtractor;
+
+    $propertyInfo = new PropertyInfoExtractor(
+        $arrayOfListExtractors,
+        $arrayOfTypeExtractors,
+        $arrayOfDescriptionExtractors,
+        $arrayOfAccessExtractors
+    );
+
+The order of extractor instances within an array matters, as the first non-null result will be returned.
+That is why you must provide each category of extractors as a separate array, even if an extractor provides information
+for more than one category.
+
+For example, while the :class:`ReflectionExtractor<Symfony\\Component\\PropertyInfo\\Extractor\\ReflectionExtractor>`
+and :class:`DoctrineExtractor<Symfony\\Bridge\\Doctrine\\PropertyInfo\\DoctrineExtractor>` both provide list and type
+information it is probably better that:
+
+* The :class:`ReflectionExtractor<Symfony\\Component\\PropertyInfo\\Extractor\\ReflectionExtractor>` has priority for
+  list information so that all properties in a class (not just mapped properties) are returned.
+* The :class:`DoctrineExtractor<Symfony\\Bridge\\Doctrine\\PropertyInfo\\DoctrineExtractor>` has priority for type
+  information so that entity metadata is used instead of type-hinting to provide more accurate type information.
+
+.. code-block:: php
+
+    use Symfony\Bridge\Doctrine\PropertyInfo\DoctrineExtractor;
+    use Symfony\Component\PropertyInfo\Extractor\ReflectionExtractor;
+    use Symfony\Component\PropertyInfo\PropertyInfoExtractor;
+
+    $reflectionExtractor = new ReflectionExtractor();
+    $doctrineExtractor = new DoctrineExtractor(/* ... */);
+
+    $propertyInfo = new PropertyInfoExtractor(
+        // List extractors
+        array(
+            $reflectionExtractor,
+            $doctrineExtractor
+        ),
+        // Type extractors
+        array(
+            $doctrineExtractor,
+            $reflectionExtractor
+        )
+    );
+
+.. _`components-property-information-usage-configuration`:
+
+Configuration
```

Comment:
the component docs shouldn't document framework integration. This has to be added to the config reference